eth/backend.go

```diff
@@ -538,3 +546,34 @@ func (s *Ethereum) Stop() error {
 
 	return nil
 }
+
+// checkStateCompatibility ensures the requested state scheme is compatible with
+// stored state data.
+func (s *Ethereum) checkStateCompatibility() error {
```

Comment:
Could we use this to detect what's on disk and default to that if none is specified?

Reply:
good idea. I will do it.